### PR TITLE
Add `ondelete` and `onupdate` attributes to macaroon warning table

### PR DESCRIPTION
**Pull request by facutuesca**
This should fix the errors when deleting a Project that has an entry in the `project_macaroon_warning_association` table.
cc @di @woodruffw

### CODE REVIEW

File: warehouse/packaging/models.py [hunk 1]
```diff
@@ -901,4 +901,7 @@ class ProjectMacaroonWarningAssociation(db.Model):
         ForeignKey("macaroons.id", onupdate="CASCADE", ondelete="CASCADE"),
         primary_key=True,
     )
-    project_id = mapped_column(ForeignKey("projects.id"), primary_key=True)
+    project_id = mapped_column(
+        ForeignKey("projects.id", onupdate="CASCADE", ondelete="CASCADE"),
+        primary_key=True,
+    )
```
Comment (woodruffw) [inline warehouse/packaging/models.py:907]: LGTM -- let's add a unit test to backstop this behavior 🙂

Comment (facutuesca) [reply warehouse/packaging/models.py:907]: done!

**Review by woodruffw — APPROVED** [review]
Thanks @facutuesca!

**Comment by woodruffw** [comment]
(Needs rebase/merge but otherwise looks G2G.)

**Review by di — APPROVED** [review]

**Pull request merged**